crates/adapters/src/transport/datagen.rs

```diff
@@ -650,7 +770,8 @@ impl RecordGenerator {
         if let Value::String(str) = obj {
             str.clear();
 
-            let (min, max) = settings.range.unwrap_or((0, MAX_DATE_VALUE));
+            let range = parse_range_for_date(field.name.as_str(), &settings.range)?;
+            let (min, max) = range.unwrap_or((0, MAX_DATE_VALUE));
```

Comment:
is 0 1970?
What if the range starts earlier?

Reply:
this is the default range if the user doesn't provide one

Reply:
but will it work with negative values?